List of global item variables (#4)
Miepee/AM2R-Modding-Wiki: content/global items/_index.md

---
title: Global Items
---

> Global item variables are variables that keep track of what the player has and what the player has equipped. Adding new items and simply setting additional items such as Screw Attack will be much easier when you find out what global variable corresponds to what item.

Reviewer: ```suggestion
Global item variables are variables that keep track of what items the player has collected and equipped. Adding new items or replacing existing items such as Screw Attack will be much easier when you know which global variable corresponds to what item.
```

> There are two separate variables for most items that either keep track of if the player has obtained it or if the player has it equipped. For example, Spider Ball has `global.item[2]` which `1` indicates that the player has it. `global.spiderball`, however, actually dictates if the player is able to use it or not. If `global.spiderball` is equivalent to `0` in this case, the player has it and can enable it, but it is currently disabled.

Reviewer: ```suggestion
There are two separate variables for most items that either keep track of whether the player has obtained it or whether the player has it currently equipped. Whether items have been obtained is stored in an `global.item` array, while whether the item is currently enabled is stored in an per-item individually named variable.  
For example, Spider Ball has `global.item[2]` which when set to `1` indicates that the player has obtained it. `global.spiderball`, however, actually determines if the player is able to use it or not. If `global.spiderball` is set to `0` but `global.item[2]` is set to `1`, then the player has it and can enable it, but it is currently disabled.
```

> Basically, `global.item` makes it so it is able to be turned on/off and the other variable, depending on its name, will actually mean the power up is active.

Reviewer: ```suggestion

```
Unnecessary with the change from above.

>  For setting them up as temporary powerups you may only want to set the player having them active, but not unlocked, then set it as inactive once the powerup has expired.

Reviewer: ```suggestion
If you plan to set up items as temporary powerups, you may want to set your item where it sets itself as active, but not unlocked, then sets itself as inactive when you want the temporariness to end.
```

> | Global Item Variable | Active Item Variable | Corresponding Item |

Reviewer: ```suggestion
| Global Item Variable | Currently Active Item Variable | Corresponding Item |
```
Alternatively `Enabled Item Variable`

| :------------------: | :------------------: | :----------------: |
| global.item[0] | global.bomb | Bombs |
| global.item[1] | global.powergrip | Power Grip |
| global.item[2] | global.spiderball | Spider Ball |
| global.item[3] | global.jumpball | Spring Ball |
| global.item[4] | global.screwattack | Screw Attack |

> | global.item[5] | global.currentsuit == 1 | Varia Suit |

Reviewer: ```suggestion
| global.item[5] | global.currentsuit (should return `1`) | Varia Suit |
```

| global.item[6] | global.spacejump | Space Jump |
| global.item[7] | global.speedbooster | Speed Booster |
| global.item[8] | global.hijump | Hi-Jump |

> | global.item[9] | global.currentsuit == 2 | Gravity Suit |

Reviewer: ```suggestion
| global.item[9] | global.currentsuit (should return `2`) | Gravity Suit |
```

| global.item[10] | global.cbeam | Charge Beam |
| global.item[11] | global.ibeam | Ice Beam |
| global.item[12] | global.wbeam | Wave Beam |
| global.item[13] | global.sbeam | Spazer Beam |
| global.item[14] | global.pbeam | Plasma Beam |
| global.item[15] | global.morphball | Morph Ball |

> Colliding with an item object will switch on both the global item variable and the active item variable, enabling the player to turn on and off the powerup and have it active on pickup.

Reviewer: ```suggestion
Colliding with an item object will switch on both the global item variable and the currently active item variable, enabling the player to turn on and off the powerup and have it enabled when picked up.
```

> It is important to note that the Power Suit is not a major item. Suits are chosen depending on what `global.currentsuit` suit is. `0` is Power Suit, `1` is Varia Suit, `2` is Gravity Suit.

Reviewer: Make this as a note please:
https://geekdocs.de/shortcodes/hints/

> Strangely, there is currently no Morph Ball item that grants morph ball, but there is a Power Grip item, despite both already being obtained and active on game start.

Reviewer: ```suggestion
Despite Morph Ball and Power Grip being obtained and enabled on game start, there exists a Power Grip item, but no Morph Ball item. This is a leftover from Metroid:Confrontation.
```

---
title: Global Items
---

> Global item variables are global variables that keep track of what the items player has collected. These variables are list items in the `gobal.item[]` array. Adding new items or replacing existing items such as Screw Attack will be much easier when you know which global variable corresponds to what item.

Reviewer: ```suggestion
Item variables are global variables that keep track of what items the player has collected and enabled. These variables are listed in the `gobal.item[]` array. Adding new items or replacing existing items such as Screw Attack will be much easier when you know which global variables corresponds to what item.
```

> There are two separate variables for most items that either keep track of if the player has obtained it or if the player has it equipped. For example, Spider Ball has `global.item[2]` which `1` indicates that the player has it. `global.spiderball`, however, actually dictates if the player is able to use it or not. If `global.spiderball` is equivalent to `0` in this case, the player has it and can enable it, but it is currently disabled.

Reviewer: ```suggestion
There are two separate variables for most items that either keep track of whether the player has it obtained or equipped. For example, Spider Ball is mapped to `global.item[2]` which if set to `1` indicates that the player has collected it. `global.spiderball`, however, dictates if the player is able to use it or not. If `global.spiderball` is equivalent to `0`, then in this case, the player has the item collected and can enable it, but it is currently disabled.
```

{{< hint type="note">}}

> The variable tracking if a major is currently equipped is not part of the global item array.

Reviewer: ```suggestion
The variable tracking of whether a major is currently equipped or not is not part of the `global.item` array.
```

{{< /hint >}}

>  For setting up majors as temporary powerups you may only want to set the player having them active, but not unlocked, then set it as inactive once the powerup has expired.

Reviewer: ```suggestion
If you want to set up majors as temporary powerups you should set the item to be active but not unlocked, and then set it as inactive once the powerup has expired.
```

| Global Item Variable | Currently Active Item Variable | Corresponding Item |
| :------------------: | :------------------: | :----------------: |
| global.item[0] | global.bomb | Bombs |
| global.item[1] | global.powergrip | Power Grip |
| global.item[2] | global.spiderball | Spider Ball |
| global.item[3] | global.jumpball | Spring Ball |
| global.item[4] | global.screwattack | Screw Attack |
| global.item[5] | global.currentsuit (should return `1`) | Varia Suit |
| global.item[6] | global.spacejump | Space Jump |
| global.item[7] | global.speedbooster | Speed Booster |
| global.item[8] | global.hijump | Hi-Jump |
| global.item[9] | global.currentsuit (should return `2`) | Gravity Suit |
| global.item[10] | global.cbeam | Charge Beam |
| global.item[11] | global.ibeam | Ice Beam |
| global.item[12] | global.wbeam | Wave Beam |
| global.item[13] | global.sbeam | Spazer Beam |
| global.item[14] | global.pbeam | Plasma Beam |
| global.item[15] | global.morphball | Morph Ball |

{{< hint type="note">}}

> Colliding with an item object will switch on both the global item variable and the currently active item variable, enabling the player to turn on and off the powerup and have it enabled when picked up.

Reviewer: ```suggestion
Colliding with an item object will switch on both the collected item variable and the currently active item variable, enabling the player to turn the powerup on and off and have it enabled when picked up.
```

{{< /hint >}}

It is important to note that the Power Suit is not a major item. Suits are chosen depending on what `global.currentsuit` suit is. `0` is Power Suit, `1` is Varia Suit, `2` is Gravity Suit.

> The game assumes the player always has the Morph Ball no matter what, mainly because it is already acquired on the start of the game.
> Despite Morph Ball and Power Grip being obtained and enabled on game start, there exists a Power Grip item, but no Morph Ball item. This is a leftover from Metroid:Confrontation.

Reviewer: ```suggestion
Despite Morph Ball and Power Grip being obtained and enabled on game start, there exists a Power Grip item, but no Morph Ball item. This is a leftover from Metroid:Confrontation. However, the game assumes the player always has the Morph Ball no matter what, mainly because it is already acquired on the start of the game.
```

{{< hint type="note">}}
Metroid:Confrontation is the tech demo for AM2R, featuring a different storyline and less developed gameplay features.

> {{< /hint >}}

Reviewer: This can be removed.
